src/main/java/ecommerce/dao/impl/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,184 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get user with login " + login, e);
+        }
+    }
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO users (name, login, password) VALUES (?, ?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.executeUpdate();
+            try (ResultSet resultSet = statement.getGeneratedKeys()) {
+                resultSet.next();
+                user.setId(resultSet.getLong(1));
+                statement.close();
+                addRolesToUser(user, connection);
+                return user;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create user" + user.toString(), e);
+        }
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        String query = "SELECT * FROM users WHERE user_id = ? AND deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    User user = getUserFromResultSet(resultSet);
+                    user.setRoles(getRolesByUserId(user.getId(), connection));
+                    return Optional.of(user);
+                }
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get use with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<User> getAll() {
+        String query = "SELECT * FROM users WHERE deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query);
+                 ResultSet resultSet = statement.executeQuery()) {
+            List<User> users = new ArrayList<>();
+            while (resultSet.next()) {
+                User user = getUserFromResultSet(resultSet);
+                user.setRoles(getRolesByUserId(user.getId(), connection));
+                users.add(user);
+            }
+            return users;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all users", e);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        String query = "UPDATE users SET name = ?, login = ?, password = ? "
```

Comment:
what if user has new role added or deleted, in your case that will not be updated

Reply:
fixed